src/diag.rs

```diff
@@ -187,7 +187,7 @@ fn annotate_diagnostic(notes: &mut Vec<Notation>,
     }
 
     fn t(v: &Token) -> String {
-        String::from_utf8((**v).to_owned()).expect("utf-8 is checked before making Token")
+        unsafe { String::from_utf8_unchecked((**v).to_owned()) }
```

Comment:
Most `Token` and `TokenPtr` which are constructed never get converted to `&str`, though, so checking should be deferred to conversion time IMO.

I guess I don't understand the motivation?  It can't be a runtime optimization, since that function is called zero times in a typical run; is it trying to shave bytes off the binary?